src/main/java/com/firebolt/jdbc/service/FireboltStatementService.java

```diff
@@ -63,11 +61,11 @@ public boolean isStatementRunning(String statementLabel) {
 		return statementClient.isStatementRunning(statementLabel);
 	}
 
-	private FireboltResultSet createResultSet(InputStream inputStream, QueryRawStatement initialQuery, FireboltProperties properties, FireboltStatement statement, int maxRows, int maxFieldSize)
+	private FireboltResultSet createResultSet(InputStream inputStream, QueryRawStatement initialQuery, FireboltProperties properties, FireboltStatement statement)
 			throws SQLException {
 		return new FireboltResultSet(inputStream, Optional.ofNullable(initialQuery.getTable()).orElse(UNKNOWN_TABLE_NAME),
 				Optional.ofNullable(initialQuery.getDatabase()).orElse(properties.getDatabase()),
-				properties.getBufferSize(), maxRows, maxFieldSize, properties.isCompress(), statement,
+				properties.getBufferSize(), statement.getMaxRows(), statement.getMaxFieldSize(), properties.isCompress(), statement,
 				properties.isLogResultSet());
```

Comment:
Can this solution be extended to `FireboltResultSet`? We're doing the same thing, passing statement but also max rows and field size.

Reply:
done

Reply:
Does this mean we can remove the old constructor? https://github.com/firebolt-db/jdbc/pull/389/files#diff-62a56c126996c1cf26d2c0ef1e090463acd6ffc55d127af126857ea350507147R124-R126

Reply:
I've reduced access modifiers of some constructors. Unfortunately removing some constructors requires massive changes in tests. I will open separate issue for this. Otherwise the changes will look irrelevant for this PR.